src/els_code_navigation.erl

```diff
@@ -76,19 +76,29 @@ find([], _Kind, _Data) ->
 find([Uri|Uris0], Kind, Data) ->
   case els_dt_document:lookup(Uri) of
     {ok, [Document]} ->
-      POIs = els_dt_document:pois(Document, [Kind]),
-      case [POI || #{id := Id} = POI <- POIs, Id =:= Data] of
-        [] ->
-          find(lists:usort(include_uris(Document) ++ Uris0), Kind, Data);
-        Definitions ->
-          {ok, Uri, hd(els_poi:sort(Definitions))}
-      end;
+      find_in_document([Uri|Uris0], Document, Kind, Data);
     {ok, []} ->
       find(Uris0, Kind, Data)
   end;
 find(Uri, Kind, Data) ->
   find([Uri], Kind, Data).
 
+-spec find_in_document(uri() | [uri()], els_dt_document:item(), poi_kind()
+                      , any()) ->
+        {ok, uri(), poi()} | {error, any()}.
+find_in_document([Uri|Uris0], Document, Kind, Data) ->
+  POIs = els_dt_document:pois(Document, [Kind]),
+  case [POI || #{id := Id} = POI <- POIs, Id =:= Data] of
+    [] ->
+      case maybe_imported(Uri, Document, Kind, Data) of
+        {ok, U, P} -> {ok, U, P};
+        {error, not_found} ->
+          find(lists:usort(include_uris(Document) ++ Uris0), Kind, Data)
+      end;
+    Definitions ->
+      {ok, Uri, hd(els_poi:sort(Definitions))}
+  end.
+
```

Comment:
Note: this was only split out because elvis complained about too much indentation otherwise.